### ISSUE TITLE
Before and After

### ISSUE THREAD

**Issue by abourget** [onsi/ginkgo#214]
Mocha has `.before` and `.after`, which runs only once in a given `Context` or `Describe` .. that would be useful for some of our use cases.

Right now, what we do is:

```
Context("blah", func() {
  var once sync.Once
  BeforeEach(func() {
    once.Do(func() {
      fmt.Println("This will be done once, before the first .It block in this context, and only once")
    })
  })
})
```

but a similar solution for `After` would be difficult, as we don,t know if we're the last, if we're quitting the `Context` or `Describe` block.

Having something native would be great.. How hard do you think it would be to implement. I ask because I might want to take a bite at it.

**Comment by robdimsdale** [comment]
Does this pattern work well with randomization of tests? Could you move this logic to a `BeforeSuite/AfterSuite` which is guaranteed to run before/after all tests

**Comment by onsi** [comment]
This comes up from time to time.  Here is the original discussion:
https://github.com/onsi/ginkgo/issues/70

**Comment by robdimsdale** [comment]
@abourget I'm going to close this issue. If you feel you want to continue this discussion feel free to open another issue.
